### PR TITLE
Add \DateTime deserialization support

### PR DESCRIPTION
**Pull request by sroze**
This PR add `DateTime` objects deserialization.

### CODE REVIEW

File: JsonLd/ContextBuilder.php [hunk 1]
```diff
@@ -69,11 +75,34 @@ public function buildContext(Resource $resource = null)
 
             foreach ($attributes as $attributeName => $attribute) {
                 if (isset($attribute->getTypes()[0]) && 'object' === $attribute->getTypes()[0]->getType()) {
-                    $context[$attributeName] = ['@type' => '@id'];
+                    $typeClass = $attribute->getTypes()[0]->getClass();
+                    $type = '@id';
```
Comment (dunglas) [inline JsonLd/ContextBuilder.php]: Move it in the `else` block and it ill avoid an useless assignment.

File: JsonLd/ContextBuilder.php [hunk 2]
```diff
@@ -69,11 +75,34 @@ public function buildContext(Resource $resource = null)
...
             }
         }
 
         return $context;
     }
+
+    /**
+     * Guess native class type.
+     *
+     * @param $class
```
Comment (dunglas) [inline JsonLd/ContextBuilder.php]: Missing type.

File: JsonLd/ContextBuilder.php [hunk 3]
```diff
@@ -69,11 +75,34 @@ public function buildContext(Resource $resource = null)
...
+     *
+     * @return null|string
+     */
+    private function guessNativeType($class)
+    {
+        if ('DateTime' === $class) {
+            return self::XML_NS.'dateTime';
+        }
+
+        return null;
```
Comment (dunglas) [inline JsonLd/ContextBuilder.php]: Useless can be removed.

File: Serializer/JsonLdNormalizer.php [hunk 4]
```diff
@@ -251,46 +252,56 @@ public function denormalize($data, $class, $format = null, array $context = [])
...
                         }
 
-                        $value = $collection;
-                    } elseif (is_string($value)) {
-                        $value = $this->dataManipulator->getObjectFromUri($value);
+                        $collection[] = $this->dataManipulator->getObjectFromUri($uri);
+                    }
+
+                    $value = $collection;
+                } elseif (null === $this->resources->getResourceForEntity($types[0]->getClass())) {
```
Comment (dunglas) [inline Serializer/JsonLdNormalizer.php]: `null ===` can be removed.

Comment (sroze) [inline Serializer/JsonLdNormalizer.php]: Nop, it's the case where there's no resource for the given class.

File: features/relation.feature [hunk 5]
```diff
@@ -79,5 +81,4 @@ Feature: Relations support
         }
       ]
     }
-
     """
```
Comment (dunglas) [inline features/relation.feature]: Missing last blank line.

File: Serializer/JsonLdNormalizer.php (hunk 4 above)
Comment (dunglas) [inline Serializer/JsonLdNormalizer.php]: Sorry I wanted to say: `null ===` can be replaced by `!`.

Comment (sroze) [inline Serializer/JsonLdNormalizer.php]: Why not, done ;-)

**Comment by dunglas** [comment]
Thanks!

**Pull request merged**